src/server.c

```diff
@@ -5938,20 +5938,35 @@ void dismissMemory(void* ptr, size_t size_hint) {
     /* madvise(MADV_DONTNEED) can not release pages if the size of memory
      * is too small, we try to release only for the memory which the size
      * is more than half of page size. */
-    if (size && size <= server.page_size/2) return;
+    if (size_hint && size_hint <= server.page_size/2) return;
 
     zmadvise_dontneed(ptr);
 }
 
 void dismissClientMemory(client *c) {
-    listIter li;
-    listNode *ln;
-    listRewind(c->reply, &li);
-    while((ln = listNext(&li))) {
-        void *bulk = listNodeValue(ln);
-        /* Default bulk size is 16k, actually it has extra data, maybe it
-         * occupies 20k according to jemalloc bin size if using jemalloc. */
-        if (bulk) dismissMemory(bulk, ((clientReplyBlock*)bulk)->size);
+    /* Dismiss client query buffer. */
+    dismissSds(c->querybuf);
+    dismissSds(c->pending_querybuf);
+    dismissMemory(c->argv, c->argc*sizeof(robj*));
+    if (c->argc && c->argv_len_sum/c->argc >= server.page_size) {
+        for (int i = 0; i < c->argc; i++) {
+            /* c->argv[i] always is string object, the argument
+             * 'dump_size' is not used in dismissObject. */
+            dismissObject(c->argv[i], 0);
+        }
+    }
```

Comment:
you must dismiss the argv itself after freeing the items in the array 8-)

Reply:
oh, ashamed. Thanks